internal/homing/homing.go

```diff
@@ -0,0 +1,117 @@
+package homing
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/pkg/errors"
+)
+
+const (
+	HomeUrl      = "localhost:9022"
+	stopWaitTime = 5 * time.Second
+)
+
+var ipEndpoints = []string{
+	"https://checkip.amazonaws.com/",
+	"https://ipinfo.io/ip",
+	"https://api.ipify.org/",
+}
+
+func New(svc, version string, homingLogger logger.Logger, cancel context.CancelFunc) *homingService {
+	return &homingService{
+		serviceName:  svc,
+		version:      version,
+		homingLogger: homingLogger,
+		cancel:       cancel,
+	}
+}
+
+type homingService struct {
+	serviceName  string
+	version      string
+	homingLogger logger.Logger
+	cancel       context.CancelFunc
+}
+
+func (hs *homingService) CallHome(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			hs.Stop()
+		default:
+			var data telemetryData
+			var err error
+			data.Service = hs.serviceName
+			data.Version = hs.version
+			data.LastSeen = time.Now()
+			for _, endpoint := range ipEndpoints {
+				data.IpAddress, err = getIP(endpoint)
```

Comment:
Will there be a case that the ips are different?

Reply:
Not really, just have multiple endpoints in case one goes offline. The IP address is obtained from the first endpoint to deliver a response without errors